packages/config/src/store/parseStoreConfig.ts

```diff
@@ -70,6 +70,8 @@ export interface TableConfig<
   storeArgument?: boolean;
   /** Include a data struct and methods for it. Default is false for 1-column tables; true for multi-column tables. */
   dataStruct?: boolean;
+  /** Generate only `setEphemeral` which emits an event without writing to storage. Default is false. */
+  ephemeral?: boolean;
```

Comment:
Hmm, rather than reducing the schema to just one `setEphemeral` field, I wonder if we can support any/all fields, with their underlying implementation calling out to `setEphemeralRecord`?

A big reason I was initially excited by the ephemeral table design is that it could look/act like a regular table (thus multiple fields), except only emit events for records rather than all the write methods.

Reply:
The problem is probably that my comment was misleading, ephemeral option does allow multiple fields, it just removed ass `set` functions, and adds a new `setEphemeral`, which functions like `setRecord` but with a different event and no storage writes. `setEphemeral` supports the same variety/number of fields as `setRecord`

Reply:
If we go with `emit` (https://github.com/latticexyz/mud/pull/665#discussion_r1183035199) we could generate all `set*` with `emit*` and change the comment to something like "Generate a table that only emits `SetEphemeralRecord` events but doesn't write to storage."